packages/lib/src/toolbar/controls/ExportEntry.tsx

```diff
@@ -0,0 +1,63 @@
+import styles from './Selector/Selector.module.css';
+
+interface Props {
+  format: string;
+  url: URL | (() => Promise<URL | Blob>) | undefined;
+}
+
+function ExportEntry(props: Props) {
+  const { format, url } = props;
+  const label = `Export to ${format.toUpperCase()}`;
+  const filename = `data.${format}`;
+
+  async function handleBtnClick(getURLOrBlob: () => Promise<URL | Blob>) {
+    const anchor = document.createElement('a');
+    anchor.download = filename;
+    document.body.append(anchor);
```

Comment:
To start a download for which you don't know the URL right away, the idea is to add a temporary link to the DOM (invisibly), set its `href` once the URL is known, and click on it programmatically.

Reply:
Yep :+1: